src/linear_algebra/basic.lean

```diff
@@ -893,6 +898,21 @@ variables [module R M] [module R₂ M₂] [module R₃ M₃]
 include R
 open submodule
 
+section restrict
+
+variables [add_comm_monoid M₁] [module R M₁]
+
+/--
+Alternative version of `restrict` to restrict domain and codomain
+of a linear map.
+-/
+@[simps] def restrict' (f : M →ₗ[R] M₁)
+  {p : submodule R M} {q : submodule R M₁} (hf : p.map f ≤ q) :
+  p →ₗ[R] q :=
+f.restrict (λ x hx, hf ⟨x, hx, rfl⟩)
```

Comment:
Is this really nessecary? You can always use `[submodule.map_le_iff_le_comap](https://leanprover-community.github.io/mathlib_docs/linear_algebra/basic.html#submodule.map_le_iff_le_comap)` to get `restrict` from `restrict'` and vice versa.

Reply:
It is not *necessary*. Personally, I feel like this formulation is just cleaner as it is entirely in terms of linear maps and submodules. However, I can remove it if that's just my opinion :)

Reply:
You can change the argument of `linear_map.restrict` to `p ≤ q.comap f`, which should be (close to) defeq to the current one, and does not involve elements as well.

Reply:
Yes, this seems sensible!